src/features/session_replay/shared/stylesheet-evaluator.js

```diff
@@ -0,0 +1,84 @@
+import { originals } from '../../../common/config/config'
+import { isBrowserScope } from '../../../common/constants/runtime'
+
+class StylesheetEvaluator {
+  #evaluated = new WeakSet()
+  #fetchProms = []
+  /**
+  * Flipped to true if stylesheets that cannot be natively inlined are detected by the stylesheetEvaluator class
+  * Used at harvest time to denote that all subsequent payloads are subject to this and customers should be advised to handle crossorigin decoration
+  * */
+  invalidStylesheetsDetected = false
+  failedToFix = false
+
+  /**
+   * this works by checking (only ever once) each cssRules obj in the style sheets array. The try/catch will catch an error if the cssRules obj blocks access, triggering the module to try to "fix" the asset`. Returns the count of incomplete assets discovered.
+   * @returns {Number}
+   */
+  evaluate () {
+    let incompletes = 0
+    if (isBrowserScope) {
+      for (let i = 0; i < Object.keys(document.styleSheets).length; i++) {
```

Comment:
```suggestion
      for (let i = 0; i < document.styleSheets.length; i++) {
```

Reply:
nice, didnt notice that obj had that property added